src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService {
+    @Override
+    public List getAll(Class type) {
+        if (type.equals(Bulldozer.class)) {
+            return new BulldozerProducer().get();
+        } else if (type.equals(Excavator.class)) {
+            return new ExcavatorProducer().get();
+        } else if (type.equals(Truck.class)) {
+            return new TruckProducer().get();
+        }
```

Comment:
It's better to create class-level fields with producers.
It's not good to create a new instance of Producers every time

return bulldozerProducer().get();